src/parser/xasl_generation.c

```diff
@@ -2744,7 +2744,8 @@ pt_split_attrs (PARSER_CONTEXT * parser, TABLE_INFO * table_info, PT_NODE * pred
 	      node->next = NULL;
 
 	      ref_node = mq_get_references (parser, node, table_info->class_spec);
-	      pred_nodes = parser_append_node (ref_node, pred_nodes);
+	      ref_node_copy = parser_copy_tree (parser, ref_node);
+	      pred_nodes = parser_append_node (ref_node_copy, pred_nodes);
```

Comment:
It looks like 'test_sql' wasn't performing properly. After testing, you should check the results.

In my opinion, the result value of mq_get_references(), ref_node, can result in multiple nodes.
``` a.col1 + a.col2 = 3 ```
In the predicate above, two nodes a.col1 and a.col2 should be returned.

The addition of unnecessary columns occurs in the code below. I am checking more.
mq_reset_ids_and_references_helper()
```
  if (get_spec_referenced_attr)
     {
       (void) parser_walk_tree (parser, spec, mq_get_references_node, spec, pt_continue_walk, NULL);
     }
```